src/util-logopenfile.h

```diff
@@ -114,6 +114,9 @@ typedef struct LogFileCtx_ {
     /* flag to avoid multiple threads printing the same stats */
     uint8_t flags;
 
+    /* flag to store the options */
+    uint8_t options_flags;
```

Comment:
I've just checked the code by removing this and trying to find a structure different from LogFileCtx that could host that. Result is that the LogFileCtx is the first common structure that can be used in all the CreateJSONHeader call contexts.
So I see the following possibilities:
* We add a union to contain per output type options and variable in LogFileCtx
* We rework all JSON output module to use a intermediate structure that will have a pointer in structure like LogDnsFileCtx

On last point, if we look at DNS we have:
```
typedef struct LogDnsFileCtx_ {
    LogFileCtx *file_ctx;
    uint64_t flags; /** Store mode */
} LogDnsFileCtx;

typedef struct LogDnsLogThread_ {
    LogDnsFileCtx *dnslog_ctx;
    /** LogFileCtx has the pointer to the file and a mutex to allow multithreading */
    uint32_t dns_cnt;

    MemBuffer *buffer;
} LogDnsLogThread;
```
So we could have a 'wrapper' JSONLogFileCtx in LogDnsFileCtx that contain the LogFileCtx and the JSON options.